fcrepo4/fcrepo4
# improved rollback logic

**JIRA Ticket**: https://jira.lyrasis.org/browse/FCREPO-3350

# What does this Pull Request do?

1. Exceptions during rollbacks are logged; not rethrown
2. Staging directories are always cleaned up
3. General logic and logging cleanup 

# How should this be tested?

Start F6:

```
mvn -Dfcrepo.home=target/fcrepo -Dfcrepo.log=DEBUG clean jetty:run
```

Create a resource in a transaction:

```
curl -u fedoraAdmin:fedoraAdmin -v -XPOST http://localhost:8080/rest/fcr:tx
txid=<TX_ID>
curl -u fedoraAdmin:fedoraAdmin -H"Atomic-ID: ${txid}" http://localhost:8080/rest/foo -H "Link: <http://www.w3.org/ns/ldp#NonRDFSource>; rel=type" -v -H "Content-Type: text/plain" -X PUT --data-binary "bar"
```

See the staging directory:

```
ls target/fcrepo/data/staging
```

Commit the transaction:

```
curl -u fedoraAdmin:fedoraAdmin -X PUT -v http://localhost:8080/rest/fcr:tx/${txid}
```

Verify the staging directory is now gone.

Do the same steps a second time, but, instead of committing the transaction, wait for it to expire and the confirm that the staging directory was removed.

# Interested parties

@fcrepo4/committers
